src/utils.jl

```diff
@@ -656,6 +660,31 @@ function get_fillalpha(series, i::Int = 1)
     _cycle(series[:fillalpha], i)
 end
 
+function get_markercolor(series, i::Int = 1)
+    mc = series[:markercolor]
+    mz = series[:marker_z]
+    if mz == nothing
+        isa(mc, ColorGradient) ? mc : _cycle(mc, i)
+    else
+        cmin, cmax = get_clims(series[:subplot])
```

Comment:
@daschw I can't tell exactly what you want here, but this implementation is very inefficient for big datasets, and is a cause of https://github.com/JuliaPlots/Plots.jl/issues/1682 
The reason is that e.g. in the spy example, we plot 1e6 points - then this function gets called for each individual point, which then again 1) calls `get_clims` that recalculates clims for all points of all series in the plot every time, and 2) generates and allocates a new `cgrad` for each new data point.
In the spy example (`n = 100_000; a = sprand(1:n, 1:n, ones(n)); p = spy(a); display(p)`, the `get_markercolor` function is responsible for 98% of all execution time according to the profiler. Almost all of that is spent in `ignorenan_extrema` recalculating the extrema again and again, so maybe a solution could simply be to calculate the clims once at the same time as xlims and ylims and storing them in `plotattributes`.

Reply:
To be specific, I'm talking about the version (which is not in Plots atm but shows the problem very well) where `:scatter` with `:marker_z` is used to show the matrix.

Reply:
OK, I see, thanks for noticing!
I'd say we just add the methods
```julia
get_markercolor(series, cmin, cmax, i::Int = 1)
get_markercolor(series, clims, i::Int = 1)
```
and use them whenever the clims are already known.